test/e2e/e2e_test.go

```diff
@@ -25,101 +26,128 @@ import (
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+
+	csiv1alpha1 "github.com/ceph/ceph-csi-operator/api/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 const namespace = "ceph-csi-operator-system"
 
-var _ = Describe("controller", Ordered, func() {
+var k8sClient client.Client
+var clientset *kubernetes.Clientset
+
+var _ = Describe("Ceph CSI Operator", Ordered, func() {
 	BeforeAll(func() {
-		By("installing prometheus operator")
-		Expect(utils.InstallPrometheusOperator()).To(Succeed())
+		var err error
+		cfg, err := config.GetConfig()
+		Expect(err).NotTo(HaveOccurred())
+
+		err = csiv1alpha1.AddToScheme(scheme.Scheme)
+		Expect(err).NotTo(HaveOccurred())
 
-		By("installing the cert-manager")
-		Expect(utils.InstallCertManager()).To(Succeed())
+		k8sClient, err = client.New(cfg, client.Options{Scheme: scheme.Scheme})
+		Expect(err).NotTo(HaveOccurred())
+
+		clientset, err = kubernetes.NewForConfig(cfg)
+		Expect(err).NotTo(HaveOccurred())
 
 		By("creating manager namespace")
-		//nolint:gosec
-		cmd := exec.Command("kubectl", "create", "ns", namespace)
-		_, _ = utils.Run(cmd)
+		ns := &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: namespace}}
+		err = k8sClient.Create(context.TODO(), ns)
+		Expect(err).NotTo(HaveOccurred())
 	})
 
 	AfterAll(func() {
-		By("uninstalling the Prometheus manager bundle")
-		utils.UninstallPrometheusOperator()
-
-		By("uninstalling the cert-manager bundle")
-		utils.UninstallCertManager()
-
 		By("removing manager namespace")
-		//nolint:gosec
-		cmd := exec.Command("kubectl", "delete", "ns", namespace)
-		_, _ = utils.Run(cmd)
+		ns := &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: namespace}}
+		err := k8sClient.Delete(context.TODO(), ns)
+		Expect(err).NotTo(HaveOccurred())
 	})
 
-	Context("Operator", func() {
-		It("should run successfully", func() {
-			var controllerPodName string
+	Context("Operator Deployment", func() {
+		It("should deploy and run successfully", func() {
 			var err error
 
-			// projectimage stores the name of the image used in the example
-			var projectimage = "example.com/ceph-csi-operator:v0.0.1"
+			projectimage := "example.com/ceph-csi-operator:v0.0.1"
 
-			By("building the manager(Operator) image")
-			//nolint:gosec
+			By("building the operator image")
 			cmd := exec.Command("make", "docker-build", fmt.Sprintf("IMG=%s", projectimage))
 			_, err = utils.Run(cmd)
-			ExpectWithOffset(1, err).NotTo(HaveOccurred())
+			Expect(err).NotTo(HaveOccurred())
 
-			By("loading the the manager(Operator) image on Kind")
-			err = utils.LoadImageToKindClusterWithName(projectimage)
-			ExpectWithOffset(1, err).NotTo(HaveOccurred())
+			By("loading the operator image to the cluster")
+			err = utils.LoadImageToCluster(projectimage)
+			Expect(err).NotTo(HaveOccurred())
 
 			By("installing CRDs")
-			//nolint:gosec
 			cmd = exec.Command("make", "install")
 			_, err = utils.Run(cmd)
-			ExpectWithOffset(1, err).NotTo(HaveOccurred())
```

Comment:
Please revert this change, the `withOffest` part is crucial here to make sense of error message